src/main/java/com/mongodb/jdbc/MongoResultSet.java

```diff
@@ -1422,7 +1261,7 @@ public Blob getBlob(int columnIndex) throws SQLException {
     }
 
     protected Clob getClob(BsonValue o) throws SQLException {
-        return new SerialClob(getString(o).toCharArray());
+        return new SerialClob(Objects.requireNonNull(getString(o)).toCharArray());
```

Comment:
Objects.requireNonNull will throw if the o is null. This means that when the data is NULL then the driver will throw an exception when getClob is called.
I think that it should return Null instead.
Add the null check before converting the data and return Null if necessary
```suggestion
        if (checkNull(o)) {
            return null;
        }
        return new SerialClob(getString(o).toCharArray());
```

Reply:
Great catch!